quantities/tuples_body.hpp

```diff
@@ -33,21 +33,24 @@ struct TupleElementOrVoid<Tuple, index, true> {
   using Type = std::tuple_element_t<index, Tuple>;
 };
 
-template<template<typename> class Transform,
-         typename Tuple, std::size_t... indices>
-struct ApplyGenerator<Transform, Tuple, std::index_sequence<indices...>>
-    : not_constructible {
-  using Type = std::tuple<Transform<std::tuple_element_t<indices, Tuple>>...>;
-};
+template<template<typename...> typename Transform,
+         typename... Tuples>
+struct ApplyGenerator {
```

Comment:
Please make this a class and put everything except `Type` in a private part.

Reply:
Done.